### PR TITLE
drivers: spi: renesas: Support Renesas  RA8 SPI driver

### PR DESCRIPTION
**Pull request by duynguyenxa**
This PR is aimed to support SPI driver for the Renesas RA8 boards
The driver design is based on the using SPI_B FSP driver from hal_renesas to implement.

### CODE REVIEW

File: drivers/spi/spi_b_renesas_ra8.c
Comment (tbursztyka) [inline drivers/spi/spi_b_renesas_ra8.c:264]: extra line to be removed

File: drivers/spi/spi_b_renesas_ra8.c
Comment (ydamigos) [inline drivers/spi/spi_b_renesas_ra8.c:286]: Should we return an error?
```suggestion
		return -EINVAL;
```

File: drivers/spi/spi_b_renesas_ra8.c
Comment (ydamigos) [inline drivers/spi/spi_b_renesas_ra8.c:146]: Should we return an error?

**Review by ydamigos — CHANGES_REQUESTED** [review]

File: drivers/spi/spi_b_renesas_ra8.c
Comment (duynguyenxa) [reply drivers/spi/spi_b_renesas_ra8.c:286]: @ydamigos , This was referred from some other vendor. They also return 0 in this case
zephyr/drivers/spi/spi_esp32_spim.c
zephyr/drivers/spi/spi_nxp_s32.c
zephyr/drivers/spi/spi_ll_stm32.c
I'm not sure if this is some kind of copy mistake or not.
The Zephyr document also not clear in this case. It's described to allow tx_bufs and rx_bufs to be NULL
image

Comment (ydamigos) [reply drivers/spi/spi_b_renesas_ra8.c:286]: I would expect to return an error if both `tx_bufs` and `rx_bufs` are NULL.
@tbursztyka Could you help if we should return an error if both `tx_bufs` and `rx_bufs` are NULL.?

File: drivers/spi/spi_b_renesas_ra8.c
Comment (duynguyenxa) [reply drivers/spi/spi_b_renesas_ra8.c]: Sorry for late reply, I removed the line

File: drivers/spi/spi_b_renesas_ra8.c
Comment (tbursztyka) [inline drivers/spi/spi_b_renesas_ra8.c:78]: Error code should be -EIO and not 0 here

File: drivers/spi/spi_b_renesas_ra8.c
Comment (tbursztyka) [reply drivers/spi/spi_b_renesas_ra8.c:286]: 0 is fine I think, NULL value is valid on both parameters, and if it happens that both are set to NULL well: there is nothing to do, all is good.

**Review by tbursztyka — CHANGES_REQUESTED** [review]

**Review by ydamigos — APPROVED** [review]
LGTM

**Review by tbursztyka — APPROVED** [review]
lgtm as well

**Comment by ydamigos** [comment]
Added `DNM` label to wait for review from Renesas RA maintainers.
@soburi, @KhiemNguyenT please take a look.

File: drivers/spi/spi_b_renesas_ra8.c
Comment (soburi) [inline drivers/spi/spi_b_renesas_ra8.c:289]: This is not blocking, but
async and interrupt-driven are essentially independent.
In other words, in the case of ASYNC, asynchrony is achieved using the argument k_poll_signal, so there is essentially no problem with a polling implementation that does not use interrupts.

**Review by soburi — APPROVED** [review]

**Comment by ydamigos** [comment]
> Added `DNM` label to wait for review from Renesas RA maintainers. @soburi, @KhiemNguyenT please take a look.

Removed  `DNM` labels as one of Renesas RA maintainers approved the PR.

**Pull request merged**
